### PR TITLE
remove 'static' from ClientAsync.add_connection

### PR DESCRIPTION
**Pull request by luanshaotong**

### CODE REVIEW

File: rpc/src/client.rs [hunk 1]
```diff
@@ -188,14 +182,85 @@ impl Client {
...
+}
+
+// parse_response
+// try to get response from sequence of connections and write to callbacks
+pub async fn parse_response(
+    mut read_stream: OwnedReadHalf,
+    connection: Arc<ClientConnectionAsync>,
+    queue: Arc<CircularQueue>,
+) {
+    loop {
```
Comment (Tricky511) [inline rpc/src/client.rs:206]: I think need to `break loop`, after `remove_connection`, maybe change the `status` of the `connection` is nice

Comment (luanshaotong) [reply rpc/src/client.rs:206]: Yes! I'm not sure shell we use a lock?

Comment (Tricky511) [reply rpc/src/client.rs:206]: The `status` of `connection` already use `RwLock`

Comment (luanshaotong) [reply rpc/src/client.rs:206]: done

**Review by Tricky511 — APPROVED** [review]
LGTM

**Pull request merged**